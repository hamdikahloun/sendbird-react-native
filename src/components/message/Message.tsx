import React, {memo} from 'react';
import {UserMessage} from 'sendbird';
import {MyMessage} from './MyMessage';
import {OtherMessage} from './OtherMessage';

type MassageProps = {
  message: UserMessage;
  isMyMessage: boolean;
};
export const MessageMemorizd = ({message, isMyMessage}: MassageProps) => {
  return (
    <>
      {isMyMessage ? (
        <MyMessage message={message} />
      ) : (
        <OtherMessage message={message} />
      )}
    </>
  );
};

/**
 * Message - This UI component displays an individual preview item for each message in a list
 *
 * See:
 *
 * [BaseMessage](https://sendbird.github.io/core-sdk-javascript/module-model_baseMessage-BaseMessage.html)
 * @example
 *
 * <Message message={message} isMyMessage={message.userId === connectedUserId} />
 */
export const Message = memo(MessageMemorizd);
