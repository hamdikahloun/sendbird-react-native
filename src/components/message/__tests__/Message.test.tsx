import React from 'react';
import {render} from '@/../testing';
import {Message} from '../Message';

describe('Message', () => {
  it('should render correctly', async () => {
    const message: any = {message: ''};
    const screen = render(
      <Message message={message} isMyMessage={false} />,
    ).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
