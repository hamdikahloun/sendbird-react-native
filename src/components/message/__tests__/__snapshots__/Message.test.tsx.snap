// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "uri": "https://avatars.githubusercontent.com/u/5143408?s=200&v=4",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 24,
              "height": 48,
              "width": 48,
            },
            undefined,
          ]
        }
        testID="user-avatar"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#F9F9F9",
        "borderRadius": 15,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 16,
        "padding": 16,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0C0020",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
